modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -2801,16 +2801,19 @@ void GDScriptAnalyzer::update_dictionary_literal_element_type(GDScriptParser::Di
 void GDScriptAnalyzer::reduce_assignment(GDScriptParser::AssignmentNode *p_assignment) {
 	reduce_expression(p_assignment->assigned_value);
 
-#ifdef DEBUG_ENABLED
```

Comment:
Remove DEBUG_ENABLED so we track assignments in release.

$REVIEW does this break anything? I should test this works as expected in release builds